### DISCUSSION TITLE
Update custom node on click

### DISCUSSION THREAD

**Discussion by umerghayaz** [xyflow/xyflow#3109] [Q&A]
I want to update custom node  by clicking on it and form appears and I change that node. I don't know how  to do that can someone help me?

**Comment by bcakmakoglu** [comment]
Well, you could start by adding a `onNodeClick` handler which stores the current node in a temporary state and then pass that to a sidebar from which you update data.

[A very simplified example can be seen here.](https://codesandbox.io/s/dazzling-wright-hledhi?file=/src/Flow.tsx)

This can probably be solved a little better but it's just an example to give you an idea on how to go about it 😄
